src/features/diagnosticsProvider.ts

```diff
@@ -127,18 +127,20 @@ class DiagnosticsProvider extends AbstractSupport {
     private _subscriptions: Subscription[] = [];
     private _suppressHiddenDiagnostics: boolean;
 
-    constructor(server: OmniSharpServer, validationAdvisor: Advisor, languageMiddlewareFeature: LanguageMiddlewareFeature) {
+    constructor(server: OmniSharpServer, validationAdvisor: Advisor, languageMiddlewareFeature: LanguageMiddlewareFeature, options: OptionProvider) {
         super(server, languageMiddlewareFeature);
 
         this._analyzersEnabled = vscode.workspace.getConfiguration('omnisharp').get('enableRoslynAnalyzers', false);
         this._validationAdvisor = validationAdvisor;
         this._diagnostics = vscode.languages.createDiagnosticCollection('csharp');
         this._suppressHiddenDiagnostics = vscode.workspace.getConfiguration('csharp').get('suppressHiddenDiagnostics', true);
 
-        this._subscriptions.push(this._validateCurrentDocumentPipe
-            .asObservable()
-            .pipe(throttleTime(750))
-            .subscribe(async x => await this._validateDocument(x)));
+        if (!options.GetLatestOptions().enableLspDriver) {
```

Comment:
Is this because diagnostics are push based with LSP?

Reply:
yeah, diagnostics will be pushed on demand by the server now.